### PR TITLE
feat: add opt-in multi-organization runner support

### PR DESCRIPTION
**Pull request by guicaulada**
## Description

Add opt-in support for a control plane serving multiple GitHub organizations. With `enable_multi_org_runners = true`, webhook-driven runners register in the repository's organization, and each pool schedule can select an `org` and target size. The flag defaults to false and preserves existing behavior in both legacy and v2 configurations.

- Select organization-specific app installations for scale-up, job retry, pools, and scale-down instead of reusing a global installation ID.
- Scope runner-group caching by organization so identically named groups can have different IDs.
- Reuse existing `ghr:Owner` and `ghr:Type` tags for deregistration and orphan checks; apply idle retention independently per organization.
- Add configuration examples and document per-organization capacity limits, default pool-owner fallback, and app-installation requirements.

## Test Plan

- 376 control-plane tests passed, including mixed-organization batches, JIT group IDs, pool isolation, retry authentication, orphan checks, and disabled-mode compatibility.
- 29 mocked Terraform tests passed across legacy runners, runner-config, pool scheduling, and legacy/v2 multi-runner configuration forwarding. Tests used locally cached AWS provider 6.60.0; the existing dependency lock files are unchanged.
- TypeScript, ESLint, Prettier, Terraform validation/formatting, and `git diff --check` passed. Terraform reports existing provider deprecation warnings.
- Live AWS/GitHub deployment validation remains outstanding.

### CODE REVIEW

File: lambdas/functions/control-plane/src/pool/pool.ts [hunk 1]
```diff
@@ -36,7 +38,10 @@ export async function adjust(event: PoolEvent): Promise<void> {
   const ephemeral = yn(process.env.ENABLE_EPHEMERAL_RUNNERS, { default: false });
   const enableJitConfig = yn(process.env.ENABLE_JIT_CONFIG, { default: ephemeral });
   const disableAutoUpdate = yn(process.env.DISABLE_RUNNER_AUTOUPDATE, { default: false });
-  const runnerOwner = process.env.RUNNER_OWNER;
+  const runnerOwner = multiOrgEnabled() ? (event.org ?? process.env.RUNNER_OWNER) : process.env.RUNNER_OWNER;
+  if (multiOrgEnabled() && (!runnerOwner || !/^[a-zA-Z0-9][a-zA-Z0-9-]*$/.test(runnerOwner))) {
```
[the remark this answers is not in the document]
Comment (guicaulada) [reply lambdas/functions/control-plane/src/pool/pool.ts]: Valid finding. Fixed in [3d88795](https://github.com/github-aws-runners/terraform-aws-github-runner/commit/3d88795): the runtime guard now enforces 1–39 alphanumeric characters with single internal hyphens. It also rejects a trailing newline. Regression tests cover invalid and valid boundary values from both `event.org` and `RUNNER_OWNER`; all 45 pool tests pass.

File: modules/orchestration-providers/webhook/pool/variables.tf [hunk 2]
```diff
@@ -117,6 +119,13 @@ variable "config" {
     log_group_tags                 = optional(map(string), {})
     user_agent                     = string
   })
+
+  validation {
+    condition = !var.config.enable_multi_org_runners || alltrue([
+      for pool in var.config.pool : can(regex("^[a-zA-Z0-9][a-zA-Z0-9-]*$", pool.org == null ? var.config.runner.pool_owner : pool.org))
```
[the remark this answers is not in the document]
Comment (guicaulada) [reply modules/orchestration-providers/webhook/pool/variables.tf]: Valid finding. Fixed in [3d88795](https://github.com/github-aws-runners/terraform-aws-github-runner/commit/3d88795): this validator now rejects leading, trailing, and repeated hyphens and logins longer than 39 characters. Tests cover schedule overrides, default owners, valid boundary values, and unchanged disabled-mode behavior; all 19 webhook-pool Terraform tests pass.

File: modules/runners/pool/variables.tf [hunk 3]
```diff
@@ -79,6 +81,13 @@ variable "config" {
     lambda_tags                          = map(string)
     user_agent                           = string
   })
+
+  validation {
+    condition = !var.config.enable_multi_org_runners || alltrue([
+      for pool in var.config.pool : can(regex("^[a-zA-Z0-9][a-zA-Z0-9-]*$", pool.org == null ? var.config.runner.pool_owner : pool.org))
```
[the remark this answers is not in the document]
Comment (guicaulada) [reply modules/runners/pool/variables.tf]: Valid finding. Fixed in [3d88795](https://github.com/github-aws-runners/terraform-aws-github-runner/commit/3d88795): the legacy pool validator now enforces the same login constraints as the runtime and webhook-provider pool. Added direct legacy-module tests for invalid schedule overrides/default owners, valid boundary values, and disabled-mode compatibility; all 10 tests pass.

File: lambdas/functions/control-plane/src/pool/pool.ts [hunk 4]
```diff
@@ -36,7 +38,17 @@ export async function adjust(event: PoolEvent): Promise<void> {
   const ephemeral = yn(process.env.ENABLE_EPHEMERAL_RUNNERS, { default: false });
   const enableJitConfig = yn(process.env.ENABLE_JIT_CONFIG, { default: ephemeral });
   const disableAutoUpdate = yn(process.env.DISABLE_RUNNER_AUTOUPDATE, { default: false });
-  const runnerOwner = process.env.RUNNER_OWNER;
+  const runnerOwner = multiOrgEnabled() ? (event.org ?? process.env.RUNNER_OWNER) : process.env.RUNNER_OWNER;
```
[the remark this answers is not in the document]
Comment (guicaulada) [reply lambdas/functions/control-plane/src/pool/pool.ts]: Valid finding. Fixed in [4675ae7](https://github.com/github-aws-runners/terraform-aws-github-runner/commit/4675ae7). Multi-org mode now lowercases organization identity in pools, scale-up, retry, and scale-down/orphan processing. Capacity lookups retain EC2 environment/type filters but compare owner tags case-insensitively after pagination, so existing mixed-case tags count toward the same limit without retagging. Regression tests cover shared capacity and idle retention, registration ownership, retries, and old tags. All 403 control-plane and 307 compute-provider tests pass; disabled-mode behavior is preserved.

File: lambdas/functions/control-plane/src/scale-runners/scale-down.ts [hunk 5]
```diff
@@ -289,6 +290,9 @@ async function evaluateAndRemoveRunners(
   const ownerTags = new Set(runners.map((runner) => runner.owner));
 
   for (const ownerTag of ownerTags) {
+    if (multiOrgEnabled()) {
+      idleCounter = getIdleRunnerCount(scaleDownConfigs);
+    }
```
[the remark this answers is not in the document]
Comment (guicaulada) [reply lambdas/functions/control-plane/src/scale-runners/scale-down.ts:301]: Valid finding. Fixed in [74dfdde](https://github.com/github-aws-runners/terraform-aws-github-runner/commit/74dfdde). In multi-org mode, legacy repository runners and organization runners now share one idle allowance per normalized organization, with eviction ordering applied across both types. Each runner retains its original owner and type for installation lookup and GitHub cleanup.

Regression tests reproduce the previous excess retention and cover both eviction orders, mixed-case repository owners, independent retention for another organization, and the correct repository/organization deletion endpoints. All 134 scale-down tests, TypeScript checking, and lint passed.
